### ISSUE TITLE
Deleting .reverse attribute causes problems pickling restart file.

### ISSUE THREAD

**Issue by rwest** [ReactionMechanismGenerator/RMG-Py#822]
tl;dr: Saving restart files is broken.

To demonstrate, start with the minimal example `/examples/rmg/minimal/input.py` and modify the options block so that it saves the restart file every loop, e.g.:
`saveRestartPeriod=(1,'seconds'),` like this: [input.py.txt](https://github.com/ReactionMechanismGenerator/RMG-Py/files/596120/input.py.txt)

It will crash with something like this:
```python
Saving restart file...
Traceback (most recent call last):
  File "/Users/rwest/Code/RMG-Py/rmg.py", line 152, in <module>
    rmg.execute(**kwargs)
  File "/Users/rwest/Code/RMG-Py/rmgpy/rmg/main.py", line 636, in execute
    self.saveEverything()
  File "/Users/rwest/Code/RMG-Py/rmgpy/rmg/main.py", line 779, in saveEverything
    self.notify()
  File "/Users/rwest/Code/RMG-Py/rmgpy/util.py", line 109, in notify
    observer.update(self)
  File "/Users/rwest/Code/RMG-Py/rmgpy/restart.py", line 93, in update
    save(rmg)
  File "/Users/rwest/Code/RMG-Py/rmgpy/restart.py", line 42, in save
    delay=0 if rmg.done else rmg.saveRestartPeriod.value_si
  File "/Users/rwest/Code/RMG-Py/rmgpy/restart.py", line 65, in saveRestartFile
    cPickle.dump(reactionModel, f, cPickle.HIGHEST_PROTOCOL)
  File "/Users/rwest/Code/RMG-Py/rmgpy/data/kinetics/family.py", line 126, in __reduce__
    self.reverse,
AttributeError: 'TemplateReaction' object has no attribute 'reverse'
```

Running under a debugger I find the reaction in this case, and at least one other example I have tried, (and I hypothesize always,) is `intra_H_migration` which is its own reverse.

Looking at the `__init__` code I couldn't see how it's possible to make a `TemplateReaction` object without a `.reverse` attribute.

However, then notice that in `rmgpy/rmg/model.py` line 654 we have:
```
                         # We're done with the "reverse" attribute, so delete it to save a bit of memory
                        delattr(reaction,'reverse')
```

I suspect this, which predates the TemplateReaction pickling change in fcbb3838cd2113c4a90138cbefba7d08ddaf8f6f, is causing the problem.

One possibility is replace `delattr(reaction,'reverse')` with `reaction.reverse=None`.
Another is remove it entirely and keep the `reverse` attribute.
Another is change the pickling assumption that the reverse always exists.

Also worth considering is why this wasn't caught in any unit or regression tests:
We should be testing the restart functionality somewhere.

**Comment by rwest** [comment]
Bump. Restart is still broken.

**Comment by samimiomid** [comment]
I have the same problem and the restart is still broken. Any suggestion?

**Comment by KEHANG** [comment]
I'll look into it.

**Comment by KEHANG** [comment]
fixed by #878
